### ISSUE TITLE
Backwards Compatibility with active Ruby versions

### ISSUE THREAD

**Issue by baweaver** [bkuhlmann/gemsmith#4]
Would it be possible to make the latest branch backwards compatible to the oldest active Ruby version? (2.3.x)

I'd like to be able to ship to a reasonable subset of the community, and unfortunately 2.5.x is broken for a lot of enterprise usecases (protobuf, kafka, etc)

**Comment by baweaver** [comment]
That's fair, and I would be willing to chip in some work to help make that happen if it is an item of interest.

**Comment by bkuhlmann** [comment]
Yeah, I'm afraid not at this time. I am sorry about this but I can't afford the cost as I try to run a tight ship with my professional and open source work. Keeping all of my projects on the same version is much easier to maintain than having a couple of projects on different versions. Even with additional help from the community it increases the complexity of maintenance over time (I used to do this many years back when I used the Travis CI platform and found it was to painful to maintain). Gemsmith is a key component to all of my open source and professional work so it's hard to justify adding unnecessary complexity.

We can keep discussing this if you want but I'll have to close this issue for now.
